src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -959,6 +960,30 @@ public virtual TokenValidationResult ValidateToken(string token, TokenValidation
             if (tokenParts.Length != JwtConstants.JwsSegmentCount && tokenParts.Length != JwtConstants.JweSegmentCount)
                 return new TokenValidationResult { Exception = LogHelper.LogExceptionMessage(new ArgumentException(LogHelper.FormatInvariant(LogMessages.IDX14111, token))), IsValid = false };
 
+            if (validationParameters.ConfigurationManager != null)
+            {
+                try
+                {
+                    validationParameters.Configuration = validationParameters.ConfigurationManager.GetBaseConfigurationAsync(CancellationToken.None).GetAwaiter().GetResult();
+                }
+                catch (AggregateException ex)
+                {
+                    // If either a signing key or a valid issuer was not provided through the TVP, we want to stop processing on configuration retrieval failure.
+                    // Otherwise, we should keep going with validation.
+                    if ((string.IsNullOrWhiteSpace(validationParameters.ValidIssuer) && validationParameters.ValidIssuers == null)
```

Comment:
It is possible that delegates are available for ValidatingIssuer or Signature. ValidateIssuer may be off or require signature may be false.
We also need to consider the token may be in JWE format.
Best not to assume too much here.

Reply:
Should we check for empty validationParameters.ValidIssuers too?

Reply:
I included checks here for the delegates, the ValidateIssuer and RequireSignedTokens flags, decryption keys, and the possibility of empty collections for the signing keys, decryption keys, and valid issuers.

The logic here has gotten quite complex, trying to think of whether this complexity is warranted. 

Should we continue with validation if a decryption key is present but there is no signing key?

Reply:
I suggest we just continue and let it fail later.
It is hard to calculate all the possibilities

Reply:
@brentschmaltz Do you think including this check is still valuable for the scenarios it currently captures?

Reply:
I added a CollectionUtilities class as Danny suggested since I think it simplifies the logic here quite a bit:  https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1695/files#diff-d6203951e787f84d178b2d2ae5f8375daa4e79d7a333b2ca9ee91a48ca66423f

`cs
public static bool IsNullOrEmpty<T>(this IEnumerable<T> enumerable)
        {
            return enumerable == null || !enumerable.Any();
        }
`